src/probability.ts

```diff
@@ -0,0 +1,186 @@
+import {GameState} from "./gameState"
+import {Position} from "./position"
+
+export class Probability{
+
+    // Returns an array of valid positions, with the frequency of each position being proportional to it's "probability" calculuated by getDistribution()
+    public static getTargetArray(gamestate:GameState, grid:number, init:boolean):Array<Position>{
+        let shift:number = Probability.getOptimalShift(gamestate, grid);
+        let distribution:number[][] = Probability.getDistribution(gamestate);
+        let validTargets:Array<Position> = [];
+        for (let row:number = 0; row<10; row++){
+            for (let column:number = 0; column<10; column++){
+                let pos:Position = new Position({"Row": GameState.numberToLetter[row],"Column":column+1});
+                if (init){
+                    if (((row>7)||(row<2) && ((column>7)||(column<2)))&&(gamestate.isValidTarget(pos))){
+                        for (let frequency:number = 0; frequency < Math.floor(34/Math.sqrt(distribution[pos.row][pos.column])); frequency++){
```

Comment:
Lot of magic numbers here. (mainly 34, 2, 7, etc.)

Do you reckon you could make them constants, or derive them from 10? (The latter would future proof this as well - that way one could change the grid size and the bot would still work!